examples/storm-starter/src/jvm/storm/starter/BlobStoreAPIWordCountTopology.java

```diff
@@ -0,0 +1,288 @@
+    // storm blobstore create --file blacklist.txt --acl o::rwa key
+    private static void createBlobWithContent(String blobKey, ClientBlobStore clientBlobStore, File file)
+            throws AuthorizationException, KeyAlreadyExistsException, IOException,KeyNotFoundException {
+        String stringBlobACL = "o::rwa";
+        AccessControl blobACL = BlobStoreAclHandler.parseAccessControl(stringBlobACL);
+        List<AccessControl> acls = new LinkedList<AccessControl>();
+        acls.add(blobACL); // more ACLs can be added here
+        SettableBlobMeta settableBlobMeta = new SettableBlobMeta(acls);
+        AtomicOutputStream blobStream = clientBlobStore.createBlob(blobKey,settableBlobMeta);
+        blobStream.write(readFile(file).toString().getBytes());
+        blobStream.close();
+    }
+
+    // Equivalent update command on command line
+    // storm blobstore update --file blacklist.txt key
+    private static void updateBlobWithContent(String blobKey, ClientBlobStore clientBlobStore, File file)
+            throws KeyNotFoundException, AuthorizationException, IOException {
+        AtomicOutputStream blobOutputStream = clientBlobStore.updateBlob(blobKey);
+        blobOutputStream.write(readFile(file).toString().getBytes());
+        blobOutputStream.close();
+    }
+
+    private static String getRandomSentence() {
+        String[] sentences = new String[]{ "the cow jumped over the moon", "an apple a day keeps the doctor away",
+                "four score and seven years ago", "snow white and the seven dwarfs", "i am at two with nature" };
+        String sentence = sentences[new Random().nextInt(sentences.length)];
+        return sentence;
+    }
+
+    private static Set<String> getRandomWordSet() {
+        Set<String> randomWordSet = new HashSet<>();
+        Random random = new Random();
+        String[] words = new String[]{ "cow", "jumped", "over", "the", "moon", "apple", "day", "doctor", "away",
+                "four", "seven", "ago", "snow", "white", "seven", "dwarfs", "nature", "two" };
+        // Choosing atmost 5 words to update the blacklist file for filtering
+        for (int i=0; i<5; i++) {
+            randomWordSet.add(words[random.nextInt(words.length)]);
+        }
+        return randomWordSet;
+    }
+
+    private static Set<String> parseFile(String fileName) throws IOException {
+        File file = new File(fileName);
+        Set<String> wordSet = new HashSet<>();
+        if (!file.exists()) {
+            return wordSet;
+        }
+        StringTokenizer tokens = new StringTokenizer(readFile(file).toString(), "\r\n");
+        while (tokens.hasMoreElements()) {
+            wordSet.add(tokens.nextToken());
+        }
+        LOG.info("parseFile {}", wordSet);
+        return wordSet;
+    }
+
+    private static StringBuilder readFile(File file) throws IOException {
+        String line;
+        StringBuilder fileContent = new StringBuilder();
+        // Do not use canonical file name here
```

Comment:
I don't quite understand how the file is being found when providing the filename and not the path.  Can you please provide me with some clarification.  Thanks!